File: 프로그래머스/unrated/172928. 공원 산책/공원 산책.js
function solution(park, routes) {
    let curX = 0
    let curY = 0
    for(let i=0;i<park.length;i++){
        for(let j = 0; j<park[0].length;j++){
            if(park[i][j] == 'S'){
                curX = j
                curY = i
            }
        }
    }
    var answer = [];
    
    let dx = 0
    let dy =0;
    let xLength = park[0].length;
    let yLength = park.length;
    let nextX;
    let nextY;
    for(let i = 0;i<routes.length;i++){
    	let tempArr = routes[i].split(' ')
        if(tempArr[0] === 'E'){
            dx = 1;
            dy = 0;
        }else if(tempArr[0] === 'S'){
            dx = 0;
            dy = 1;
        }else if(tempArr[0] === 'W'){
            dx = -1;
            dy = 0;
        }else{
            dx = 0;
            dy = -1;
        } 	
        let distance = parseInt(tempArr[1]);
        nextX = curX;
        nextY = curY;
        for(let j = 0; j<distance; j++){
            console.log(nextX,nextY);
            nextX = nextX+dx;
            nextY = nextY+dy;
          
       		if(nextX <0 || nextX >= xLength || nextY <0 || nextY >= yLength || park[nextY][nextX] ==="X"){
                nextX = curX;
                nextY = curY;
                break;
            }
        }
        curX = nextX;
        curY = nextY;
    }
    answer.push(curY);
    answer.push(curX);
    
    return answer;
}

File: 프로그래머스/1/172928. 공원 산책/공원 산책.py
# 지나다니는길 O 
# 장애물 X
# 시작점 S

def solution(park, routes):
    direction = {"E":1, "S": 1, "W": -1, "N" :1}
    
    
    for i in range(len(park)):
        for j in range(len(park[i])):
            if park[i][j] == "S":
                cur_r, cur_c=(i,j)

    c,r = cur_c, cur_r
    for route in routes:
        way,distance = route.split()
        distance = int(distance)
        cur_c, cur_r = c,r
        while(distance >0):
            if way == "E":
                next_c = cur_c + 1
                next_r = cur_r
            elif way == "W":
                next_c = cur_c -1
                next_r = cur_r
            elif way == "N":
                next_r = cur_r -1
                next_c = cur_c
            else:
                next_r = cur_r + 1
                next_c = cur_c
            if (0 <= next_r <len(park)) and (0 <= next_c < len(park[next_r])) and park[next_r][next_c] != "X":
                cur_r, cur_c = next_r, next_c
            else:
                break
            distance-=1
            if distance == 0 :
                c = cur_c
                r = cur_r
            print(cur_r,cur_c)
    
    # print(start_r, start_c)
    answer = [r , c]
    return answer